src/lib/y2s390/dasds_reader.rb

```diff
@@ -0,0 +1,165 @@
+      data = dasd_entries(dasd: dasd).find { |e| e.start_with?(/\d/) }
+      return false if data.to_s.empty?
+
+      _, status, name, _, type, = data.split(" ")
+      dasd.status = status
+      dasd.device_name = name
+      dasd.type = type
+      update_additional_info(dasd) if extended
+    end
+
+    def update_additional_info(dasd)
+      refresh_extended_data!(dasd)
+    end
+
+  private
+
+    # In production, call SCR.Read(.probe.disk).
+    # For testing, point YAST2_S390_PROBE_DISK to a YAML file
+    # with the mock value.
+    # Suggesstion:
+    #   YAST2_S390_PROBE_DISK=test/data/probe_disk_dasd.yml rake run"[dasd]"
+    # @return [Array<Hash>] .probe.disk output
+    def dasd_entries(offline: true, dasd: nil)
+      if ENV["S390_MOCKING"]
+        File.read("test/data/lsdasd.txt")
+      else
+        cmd = cmd_for(offline: offline)
+        Yast::Execute.stdout.locally!(cmd)
+      end.split("\n")
+    end
+
+    def cmd_for(offline: true, dasd: nil)
+      cmd = [LIST_CMD]
+      cmd << "-a" if offline
+      cmd << dasd if dasd
+      cmd
+    end
+
+    def refresh_extended_data!(dasd)
+      reset_extended_data!(dasd) if dasd.offline?
+      dasd.use_diag = use_diag?(dasd)
+      dasd.formatted = formatted?(dasd)
+      dasd.device_type = device_type_for(dasd)
+    end
+
+    def reset_extended_data!(dasd)
+      dasd.cylinders = nil
+      dasd.formatting_status = nil
+    end
+
+    # Determines whether a given DASD is formatted or not
+    #
+    # @param dasd [DASD]
+    # @return [Boolean]
+    def formatted?(dasd)
+      return false if dasd.offline?
+
+      fmt_out = Yast::Execute.stdout.locally!(
+        ["/sbin/dasdview", "--extended", "/dev/#{dasd.device_name}"], ["grep", "formatted"]
+      )
+      fmt_out.empty? ? false : !fmt_out.upcase.match?(/NOT[ \t]*FORMATTED/)
+    end
+
+    # Determines whether the given DASD has the DIAG access enabled or not
+    #
+    # @param dasd [DASD]
+    # @return [Boolean]
+    def use_diag?(dasd)
+      diag_file = "/sys/#{dasd.sysfs_id}/device/use_diag"
+      use_diag = Yast::SCR.Read(Yast.path(".target.string"), diag_file) if File.exist?(diag_file)
+      use_diag.to_i == 1
+    end
+
+    # Get partitioninfo
+    # @param [String] disk string Disk to read info from
+    # @return GetPartitionInfo string The info
+    def partition_info(dasd)
+      return "#{dasd.device_name}1" if dasd.type != "ECKD"
+
+      out = Yast::Execute.stdout.on_target!("/sbin/fdasd", "-p", "/dev/#{dasd.device_name}")
+      return out if out.empty?
+
+      regexp = Regexp.new("^[ \t]*([^ \t]+)[ \t]+([0-9]+)[ \t]+([0-9]+)[ \t]+([0-9]+)" \
+        "[ \t]+([^ \t]+)[ \t]+([^ \t]+([ \t]+[^ \t]+))*[ \t]*$")
+
+      lines = out.split("\n").select { |s| s.match?(regexp) }
+      lines.map { |line| r = line.match(regexp);  "#{r[1]} (#{r[6]})" }.join(", ")
+    end
+
+    def device_type_for(dasd)
+      device_type = (dasd&.hwinfo&.device_id.to_i & 65535).to_s(16)
```

Comment:
For improving (just a bit) readability, I'd extract common `dast&hwinfo` and `detail`

```ruby
hwinfo = dasd&.hwinfo
detail = hwinfo&.detail

type = (hwinfo&.device_id.to_i  & 65535).to_s(16)
cu_model = detail&.cu_model.to_i.to_s(16).rjust(2, "0")


...
```